Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -106,6 +110,14 @@ public class BookmarksModel: BookmarksModelFactorySource {
     }
 
     /**
+     * Produce a new model with a memory-backed root with the given GUID removed from the current folder
+     */
+    public func removeGUIDFromCurrent(guid: GUID) -> BookmarksModel {
+        let memoryBackedFolder = MemoryBookmarkFolder.fromBookmarkFolder(self.current)
```

Comment:
I was able to reduce this and just implement the removal code in `BookmarkFolder` so at least that has defined behavior now. The only annoyance is the abstraction doesn't implement SequenceType so I just had to flatMap and run the indices manually.